rust/kernel/static_assert.rs

```diff
@@ -4,35 +4,198 @@
 
 /// Static assert (i.e. compile-time assert).
 ///
-/// Similar to C11 [`_Static_assert`] and C++11 [`static_assert`].
+/// There are several forms of this macro:
+///
+///   - Boolean assertion: `expr`.
+///   - Set membership assertion: `(expr) is in {a0, ..., aN}`.
```

Comment:
Pascal has an `in` operator. If we want to follow precedent, perhaps we could drop `is` and just have `expr in {a0, ..., aN}`.

Reply:
Yeah, Python too (as well as the `not in`). I am happy either way!